testsuite/tests/libratest/smoke_test.rs

```diff
@@ -1174,3 +1185,68 @@ fn test_malformed_script() {
         .mint_coins(&["mintb", "0", "10"], true)
         .unwrap();
 }
+
+#[test]
+fn test_key_manager_consensus_rotation() {
+    // Create and launch a local validator swarm of 2 nodes.
+    let mut swarm = TestEnvironment::new(2);
+    swarm.launch_swarm(RoleType::Validator);
+
+    // Create a node config for the key manager by modifying the first node config in the swarm.
+    // TODO(joshlind): see if we can refactor TestEnvironment to clean this up.
+    let config_path = swarm.validator_swarm.config.config_files.get(0).unwrap();
+    let mut config = NodeConfig::load(&config_path).unwrap();
+    let backend = config.consensus.safety_rules.backend.clone();
+    let json_rpc_endpoint = format!("http://127.0.0.1:{}", config.rpc.address.port());
+    config.secure.key_manager.secure_backend = backend.clone();
+    config.secure.key_manager.json_rpc_endpoint = json_rpc_endpoint.clone();
+    config.secure.key_manager.rotation_period_secs = 10;
+    config.secure.key_manager.sleep_period_secs = 10;
+    config.save(config_path).unwrap();
+
+    // Bootstrap secure storage by initializing the keys required by the key manager.
+    // TODO(joshlind): set these keys using config manager when initialization is supported.
+    let mut storage: Box<dyn Storage> = (&backend).try_into().unwrap();
+    storage
+        .create_key("consensus_previous", &Policy::default())
+        .unwrap();
+    let operator_private = config
```

Comment:
i wonder if these type of calls are frequent enough where we should make helper functions that return TestConfig, Keypairs, etc

  ```
fn test_config(&self) -> &TestConfig {
    self.test.as_ref().unwrap()
}
```

Reply:
I noticed this too.. definitely something that's worth looking into, but I'm going to punt on it for now. (I'll add it to my long list of refactors :P)